backend/core/src/main/java/org/sonarsource/sonarlint/core/analysis/AnalysisService.java

```diff
@@ -327,6 +327,7 @@ private static boolean isPrefixForAll(Path prefixCandidate, Collection<Path> pat
 
   public void setUserAnalysisProperties(String configScopeId, Map<String, String> properties) {
     userAnalysisPropertiesRepository.setUserProperties(configScopeId, properties);
+    autoAnalyzeOpenFiles(configScopeId);
```

Comment:
I've had a discussion recently about clients sometimes calling methods even if no data changed. To be on the safe side, could you re-trigger the analysis only if the properties really changed?